packages/mdc-textfield/foundation.js

```diff
@@ -233,17 +316,28 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
     }
   }
 
+  /**
+   * @return {!HTMLInputElement|!NativeInputType} The native text input from the
+   * host environment, or a dummy if none exists.
+   * @private
+   */
   getNativeInput_() {
-    return this.adapter_.getNativeInput() || {
+    return this.adapter_.getNativeInput() ||
+    /** @type {NativeInputType} */ ({
```

Comment:
Add [explicit nullability operator](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability):

`@type {!NativeInputType}`

Reply:
Done, found some more and fixed throughout.